packages/geo/src/providers/location-service/AmazonLocationServiceProvider.ts

```diff
@@ -718,19 +702,32 @@ export class AmazonLocationServiceProvider implements GeoProvider {
 	 */
 	private async _ensureCredentials(): Promise<boolean> {
 		try {
-			const credentials = await Credentials.get();
+			const credentials = (await fetchAuthSession()).credentials;
 			if (!credentials) return false;
-			const cred = Credentials.shear(credentials);
-			logger.debug('Set credentials for storage. Credentials are:', cred);
-			this._config.credentials = cred;
+			logger.debug(
+				'Set credentials for storage. Credentials are:',
+				credentials
+			);
+			this._credentials = credentials;
 			return true;
 		} catch (error) {
 			logger.debug('Ensure credentials error. Credentials are:', error);
 			return false;
 		}
 	}
 
+	private _refreshConfig() {
+		this._config = Amplify.getConfig().Geo?.LocationService;
+		if (!this._config) {
+			const errorString =
+				"No Geo configuration found in amplify config, run 'amplify add geo' to create one and run `amplify push` after";
+			logger.debug(errorString);
+			throw new Error(errorString);
```

Comment:
Might not be in scope, but in other categories we standardized on `<category>Error` which extends from Amplify error. Might be good to re-use that pattern, e.g. for auth: https://github.com/aws-amplify/amplify-js/blob/next/release/packages/auth/src/errors/AuthError.ts

```suggestion
			throw new Error(errorString);
```

Reply:
Would you be alright doing this as a follow up to this PR? Just don't want to add scope to this PR as it's been open for a while

Reply:
Yeah sounds good